app/scripts/controllers/mmi-controller.js

```diff
@@ -292,9 +292,9 @@ export default class MMIController extends EventEmitter {
       })),
     );
 
-    newAccounts.forEach(
-      async () => await this.keyringController.addNewAccount(keyring),
-    );
+    for (let i = 0; i < newAccounts.length; i++) {
```

Comment:
The forEach method does not wait for the promises returned by the addNewAccount method to resolve before moving on to the next iteration. This means that all calls to addNewAccount will be made concurrently, which could cause issues. 

TLDR; In this case, the code was continuing, so `syncAddresses` was called before `setAccountLabel`.